### DISCUSSION TITLE
Collider to multiple rigid bodies, how?

### DISCUSSION THREAD

**Discussion by ThomasCartier** [dimforge/rapier#564] [Q&A] [answered]
Hi,

I want two rigid bodies to collide with a given collider.
That collider happens to be a compound shape, but for the sake of clarity, please just look at the simplified picture:

![Screenshot_20231210_162111](https://github.com/dimforge/rapier/assets/146366386/f7fe8dcb-6fc2-4a39-9881-95cce3e2681a)

Red arrow: expected movement
Blue arrows: what happens now.

The issue is that  [attaching](https://rapier.rs/docs/user_guides/rust/colliders/#creation-and-insertion) a collider to a rigid body basically moves it 
`let handle = collider_set.insert_with_parent(collider, rigid_body_handle, &mut rigid_body_set);`

I need to attach the collider to two rigid bodies.

Cloning the collider leads to a really long simulation (I mean it) so there must be a different way to achieve this.

Right now, the two balls just fall as if the collider didn't exist.

```
let mut res = ColliderBuilder::convex_decomposition(&model_vertices, &model_indices).build();
res.set_density(1.0);

collider_set.insert(res, rb_handle1, &mut rigid_body_set);
collider_set.insert(res, rb_handle2, &mut rigid_body_set); // error
```
How to achieve this ? pretty sure it is stupid but couldn't find out.

Thanks

**Comment by sebcrozet** [comment, marked as answer]
It is not possible to attach a single collider to the same rigid-body. The code snippet your provide are actually creating two distinct colliders with the same shape, attached to two distinct rigid-body. It looks like you are trying to have the one rigid-body’s motion constrained based on the motion of another rigid-body. To achieve that, you can attach the two rigid-bodies with a constraint like a `FixedJoint`.

Another (more efficient and numerically robust) option is to only have a single rigid-body and attach the two ball colliders to that rigid-body. But of cours, this means you don’t get one rigid-body per ball (which might or might not be an issue depending on your use-case).

**Reply by ThomasCartier** [reply]
Yes, I understand. And I forgot to add colliders to the two rigids (I am an idiot).

I will apply your 2nd paragraph strategy.

Thanks a lot Seb,
